iree/tools/utils/yaml_util.c

```diff
@@ -6,11 +6,184 @@
+                            "(%zu): expected mapping node",
+                            node->start_mark.line);
+  }
+  for (yaml_node_pair_t* pair = node->data.mapping.pairs.start;
+       pair != node->data.mapping.pairs.top; ++pair) {
+    yaml_node_t* key_node = yaml_document_get_node(document, pair->key);
+    if (iree_yaml_string_equal(key_node, key)) {
+      *out_value = yaml_document_get_node(document, pair->value);
+      if (!*out_value) {
+        return iree_make_status(IREE_STATUS_INVALID_ARGUMENT,
+                                "(%zu): mapping entry has no value",
+                                key_node->start_mark.line);
+      }
+      return iree_ok_status();
+    }
+  }
+  return iree_ok_status();
+}
 
-// TODO(benvanik): yaml parsing/printing to vm types.
+iree_status_t iree_yaml_mapping_find(yaml_document_t* document,
+                                     yaml_node_t* node, iree_string_view_t key,
+                                     yaml_node_t** out_value) {
+  IREE_RETURN_IF_ERROR(
+      iree_yaml_mapping_try_find(document, node, key, out_value));
+  if (!*out_value) {
+    return iree_make_status(IREE_STATUS_NOT_FOUND,
+                            "no mapping found for key '%.*s'", (int)key.size,
+                            key.data);
+  }
+  return iree_ok_status();
+}
+
+// base64 incremental decoding with support for interior whitespace (as required
+// by the general YAML encoding convention).
+//
+// YAML !binary encoding:
+// https://yaml.org/type/binary.html
+//
+// Source:
+// https://en.wikibooks.org/wiki/Algorithm_Implementation/Miscellaneous/Base64#C_2
+static const uint8_t iree_yaml_base64_decode_table[] = {
+    66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 64, 66, 66, 66, 66, 66, 66, 66, 66,
+    66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66,
+    66, 66, 66, 66, 66, 62, 66, 66, 66, 63, 52, 53, 54, 55, 56, 57, 58, 59, 60,
+    61, 66, 66, 66, 65, 66, 66, 66, 0,  1,  2,  3,  4,  5,  6,  7,  8,  9,  10,
+    11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 66, 66, 66, 66,
+    66, 66, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42,
+    43, 44, 45, 46, 47, 48, 49, 50, 51, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66,
+    66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66,
+    66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66,
+    66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66,
+    66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66,
+    66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66,
+    66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66, 66,
+    66, 66, 66, 66, 66, 66, 66, 66, 66,
+};
+
+#define IREE_YAML_BASE64_WHITESPACE 64
+#define IREE_YAML_BASE64_EQUALS 65
+#define IREE_YAML_BASE64_INVALID 66
+
+size_t iree_yaml_base64_calculate_size(iree_string_view_t source) {
```

Comment:
I had mild hope that someday the answer wouldn't be just "write another base64 decoder". oh well...